pkg/auth/kubeconfig.go

```diff
@@ -146,24 +149,44 @@ const (
 func GenerateKubeConfig(ctx context.Context, cli kubernetes.Interface, cfg *clientcmdapi.Config, writer io.Writer, options ...KubeConfigGenerateOption) (*clientcmdapi.Config, error) {
 	opts := newKubeConfigGenerateOptions(options...)
 	if opts.X509 != nil {
-		return generateX509KubeConfig(ctx, cli, cfg, writer, opts.X509)
+		info, _ := cli.Discovery().ServerVersion()
+		if info == nil || version.MustParseGeneric(info.String()).AtLeast(version.MustParseSemantic("v1.19.0")) {
+			return generateX509KubeConfig(ctx, cli, cfg, writer, opts.X509)
+		}
+		return generateX509KubeConfigBeta(ctx, cli, cfg, writer, opts.X509)
 	} else if opts.ServiceAccount != nil {
 		return generateServiceAccountKubeConfig(ctx, cli, cfg, writer, opts.ServiceAccount)
 	}
 	return nil, errors.New("either x509 or serviceaccount must be set for creating KubeConfig")
 }
 
-func genKubeConfig(cfg *clientcmdapi.Config, authInfo *clientcmdapi.AuthInfo, caData []byte) *clientcmdapi.Config {
+func genKubeConfig(cfg *clientcmdapi.Config, authInfo *clientcmdapi.AuthInfo, caData []byte) (*clientcmdapi.Config, error) {
+	if len(cfg.Clusters) == 0 {
```

Comment:
These validation for the clientcmdapi.Config should be separated in other package. Not inside here.

Reply:
To prevent panic